### ISSUE TITLE
On the orientation of Pluto

### ISSUE THREAD

**Issue by eggrobin** [KSP-RO/RealSolarSystem#91]
The Pluto texture has its reference meridian off by 90 deg, and it is rotated upside-down.

To find the reference meridian Pluto, and the orientation of its latitude and longitude, look at [this map](https://planetcarto.files.wordpress.com/2015/07/plutomap.jpg) ((0, 0) is opposite the heart).
Side note: in this discussion it actually matters that the longitude used on Pluto yields an orientation consistent with the convention on Earth and Mars, and inconsistent with other planets and satellites of planets; happily, since we're only looking at Earth and Pluto below, we can know that it's consistent and forget about it again. For gory details on coordinate systems in planetary astronomy, see [Emily Lakdawalla's article on the matter](http://www.planetary.org/blogs/emily-lakdawalla/2013/07301442-plutosci-zangari-cartography.html#footnote).

Here is the Pluto texture currently in KSP-RO/RSS-Textures, with annotations showing the directions of latitude and longitude, and the reference meridian.
![pluto](https://cloud.githubusercontent.com/assets/2284290/17842814/0b8cf6cc-6827-11e6-88d6-08017a6e9e56.png)

Compare this with the Earth texture: the latter correctly has longitude running right to left, latitude running bottom to top, and the origin of longitude a quarter of the way in from the left.
~The biome map is similarly oriented~<sup>1</sup>, so here it is for illustration purposes; I have not found it necessary to annotate the directions of latitude and longitude nor the reference meridian, the outlines of landmasses should be familiar enough.
![earth](https://github.com/KSP-RO/RSS-Textures/raw/v13.0/8192/PluginData/EarthBiomes.png)

This wasn't caught, because, due to their inclination, Pluto and Charon cannot be tidally locked in stock. However, in the current (unreleased) state of Principia, which has axial tilt, Charon gets to look tangentially at Pluto's heart ([it should not](https://twitter.com/elakdawalla/status/625769987907547136)).

Other small solar system bodies may be affected (an incorrect rotation of anything other than the Earth and its moon is hard to spot in-game).
<hr></hr>

<sup>1</sup> The biome map and EarthColor actually differ by a reflection about the equator.

**Comment by PhineasFreak** [comment]
This is old but IIRC Kopernicus can specify the initial rotation via it's configs. For the texture rotation, it is probably because of the source image --> .dds conversion (as .dds requires all textures to be rotated vertically before export).

**Comment by eggrobin** [comment]
@PhineasFreak Yes, you can specify the initial rotation, but that is overriden by Principia's config files. Principia assumes the initial rotation is consistent with the *W* value at J2000 (`reference_angle` in https://github.com/mockingbirdnest/Principia/blob/2018021521-Cohen/astronomy/sol_gravity_model.proto.txt#L244).

It is useful to keep that convention, so that the values are consistent with those given by the WGCCRE in http://astropedia.astrogeology.usgs.gov/download/Docs/WGCCRE/WGCCRE2009reprint.pdf, rather than compensating for arbitrary transformations of the textures; otherwise when Principia adds non-rotationally symmetric gravitational fields it would become maddening to keep track of the arbitrary offsets that should be applied to textures but not gravitational fields.

Using the initial rotation to compensate for texture misalignment would also break mods that expose a longitude display, since the initial rotation rotates the body (as it should), rather than moving the origin of longitude.

The correct solution to this issue is to look at all textures and check that their orientation and reference meridians are correct, or fix them if they are not; this is tedious, I'll try to look at Pluto specifically in the near future (this obviously won't make it in the v13 release).

**Comment by PhineasFreak** [comment]
Oh right, i forgot that we can get away with these hacks only under "stock" RSS...i am sorry.

**Comment by PhineasFreak** [comment]
@eggrobin I have updated some of the Pluto textures for testing purposes (color, normal & biome maps). If/when you can, check them out: https://github.com/PhineasFreak/RSS-Textures/tree/RSS-Pluto-Rotation-Fix

I did not update the height map since i was not able to find a way to export L8 .dds textures. Either way, it is not required for this initial (feedback) round.

**Comment by eggrobin** [comment]
@PhineasFreak: sorry for the delay. Your *biome* textures in https://github.com/PhineasFreak/RSS-Textures/commit/c72e4eee87748a70c5013eb27a1504cf789684e3 look good.
Your DDS texture is oriented inconsistently with Earth's (but my statement "The biome map is similarly oriented" above is incorrect so that doesn't help :-p).

The current EarthColor.dds https://github.com/PhineasFreak/RSS-Textures/blob/c72e4eee87748a70c5013eb27a1504cf789684e3/2048/EarthColor.dds is upside-down, but *not* mirrored: longitude runs right-to-left, latitude runs bottom-to-top. This means that the PlutoColor etc. textures should be oriented as follows:
<img width="512" alt="good pluto" src="https://user-images.githubusercontent.com/2284290/41201000-3da4e8b0-6cb0-11e8-9b29-e47784329ea5.png">

Flying around Pluto and checking that the biomes match up with the landscape should confirm this.
Edit: ... or just using the biome overlay, I guess :-)

**Comment by PhineasFreak** [comment]
Weird, seems that the Pluto texture was always oriented incorrectly then, since i just added 90 degrees of offset to it. The texture should be flipped horizontally it seems.

**Comment by eggrobin** [comment]
@PhineasFreak 
> seems that the Pluto texture was always oriented incorrectly

The outlines are unfamiliar, so that makes this kind of error easy. Have you tried looking with the biome overlay to confirm?

**Comment by Growflavor** [comment]
> The correct solution to this issue is to look at all textures and check that their orientation and reference meridians are correct, or fix them if they are not; this is tedious

& also  @PhineasFreak 

If of interest to this project, adjusting textures is something I could help with as I have been trying to figure out how to identify/implement convention based cartography for myself.

[Lunaserv ](http://webmap.lroc.asu.edu/lunaserv.html )especially via [QGIS ](https://lunaserv.lroc.asu.edu/getting_started_with_lunaserv_and_qgis.pdf )makes it a bit easier now to help visualize the various coordinate systems e.g. positive pole.  (apparently from 'Emily's article on the matter' JPL HORIZONS system keeps the IAU coordinate definition for Pluto)

Io (which RSS-Textures v.13.1 seems to have as a mirror image) is my current test case as it has some verifiable [HST](https://youtu.be/K76KdULYNwg) & [Earth scope](https://youtu.be/EN0NfIlXD7k?t=5) & [probe](https://photos.app.goo.gl/fY41Gx9NBoXQwy1BA) data & [the rotation has a ~90deg difference between RSS & Principia.](https://forum.kerbalspaceprogram.com/index.php?/topic/162200-wip151-161-17x-principia%E2%80%94version-fourier-released-2019-10-28%E2%80%94n-body-and-extended-body-gravitation-axial-tilt/&do=findComment&comment=3700625) in other words the same texture is displayed differently with just RSS v RSS w/ Principia. 

In KSP 1.7.3 with Principia, the .dds 270 deg longitude centered textures for Io in the folder at the following link display as expected (e.g. HST, Voyager, SN8, NASA eyes)

https://drive.google.com/open?id=1npx6JxUbQmygotyaJNcfsIA6UH5hku4Y

I also find it interesting to see how [NASA eyes](https://eyes.nasa.gov/) & [Starry Night](https://store.simulationcurriculum.com/collections/starry-night-8/products/starry-night-pro-plus-8) display textures in those programs...though I am interested in eventually getting clear in my own head how RSS v [Principia feed Kopernicus data to rotate CBs](https://github.com/mockingbirdnest/Principia/wiki/Principia-configuration-files#the-principia_gravity_model-configuration) so I can understand each step in the process that ends up affecting the orientation of a given texture (& what might be causing RSS's configs to create a rotation ~90 deg different from Principia)

Thanks for this very useful topic & Kind regards!

**Comment by PhineasFreak** [comment]
Coming back to it after all this time, the current global map texture is wrong will be updated in a future RSS release. It will most probably incorporate the new and better textures by @dearmoon9 (https://github.com/KSP-RO/RSS-Textures/pull/23)

**Comment by PhineasFreak** [comment]
And it is now done! https://github.com/PhineasFreak/RSS-Textures-Dev/releases/tag/v0.1.0

(almost a decade later...)
